compiler_opt/distributed/local/local_worker_manager.py

```diff
@@ -114,33 +117,30 @@ class _Stub():
     """Client stub to a worker hosted by a process."""
 
     def __init__(self):
-      self._send: 'queue.Queue[Task]' = multiprocessing.get_context().Queue()
-      self._receive: 'queue.Queue[Optional[TaskResult]]' = \
-        multiprocessing.get_context().Queue()
+      parent_pipe, child_pipe = multiprocessing.get_context().Pipe()
```

Comment:
nit: can you call them incoming_pipe, outgoing_pipe? or task_result_pipe / task_pipe? (just to clarify more from the name what objects travel the pipes, which should also clarify what they are used for)

Reply:
the two vars don't exactly represent an incoming/outgoing pipe, there's just "one", bidirectional pipe. The two vars are the two ends of the pipe, .send from one corresponds to .recv on the other

Reply:
yes, but one is traversed by TaskResults, the other by Tasks, no?

Reply:
The underlying unidirectional connections, yes. 
The connections returned by .Pipe(), no- 
`parent_pipe.send(Task)`
`Optional[TaskResult] = parent_pipe.recv()`
 `child_pipe.send(TaskResult)`
 `Task = child_pipe.recv()`

Reply:
I can see how `_pipe` might be a confusing suffix though, but {incoming|outgoing|task|task_result}_pipe would be even more counter-intuitive (like `incoming_pipe.send(Task)`, `Optional[TaskResult] = task_pipe.recv()`)

The python docs just used conn1, conn2

Reply:
sorry, I probably caused the confusion with giving 4 naming choices. Let's say just: `task_pipe, task_result_pipe`. Would that cause confusion?

Reply:
still yes- task_pipe.send(Task), TaskResult = task_pipe.recv(); task_result_pipe.send(TaskResult) Task=task_result_pipe.recv(),

Reply:
ooo! I see. OK. lgtm then